File: src-tauri/src/api/user_api.rs
use std::sync::Arc;

use brainy_core::backend::{
    models::UserInformnationDto, traits::brainy_backend_client::BrainyBackendClient,
};
use tauri::State;

use crate::api::ApiError;

#[tauri::command]
pub async fn get_user_information(
    backend_client: State<'_, Arc<dyn BrainyBackendClient>>,
) -> Result<UserInformnationDto, ApiError> {
    let result = backend_client.get_user_information().await?;
    Ok(result)
}

#[tauri::command]
pub async fn update_user_information(
    backend_client: State<'_, Arc<dyn BrainyBackendClient>>,
    first_name: Option<String>,
    last_name: Option<String>,
) -> Result<(), ApiError> {
    backend_client
        .update_user_information(first_name, last_name)
        .await?;
    Ok(())
}

File: src/api/backend/api/userApi.ts
import { invoke } from "@tauri-apps/api/core";
import { UserInformationDto } from "../dto/userInformationDto";

export function getUserInformation(): Promise<UserInformationDto> {
	return invoke("get_user_information");
}

export function updateUserInformation(
	firstName: string | null,
	lastName: string | null,
): Promise<void> {
	return invoke("update_user_information", { firstName, lastName });
}

export function deleteUser(): Promise<void> {
	return invoke("delete_user");
}
